# Fast link to about:config from about
# XUL disabled about:config page, see debian/resources/branding/microb/content/config.html
Index: docshell/base/nsAboutRedirector.cpp
===================================================================
--- docshell/base/nsAboutRedirector.cpp.orig
+++ docshell/base/nsAboutRedirector.cpp
@@ -64,17 +64,21 @@
   URI.  Perhaps we should separate the two concepts out...
  */
 static RedirEntry kRedirMap[] = {
     { "credits", "http://www.mozilla.org/credits/",
       nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT },
     { "mozilla", "chrome://global/content/mozilla.xhtml",
       nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT },
     { "plugins", "chrome://global/content/plugins.html", 0 },
+#ifdef MOZ_XUL
     { "config", "chrome://global/content/config.xul", 0 },
+#else
+    { "config", "chrome://branding/content/config.html", 0},
+#endif
 #ifdef MOZ_XUL_APP
     { "logo", "chrome://branding/content/about.png",
 #else
     { "logo", "chrome://global/content/logo.gif",
 #endif
       nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT},
     { "buildconfig", "chrome://global/content/buildconfig.html",
       nsIAboutModule::URI_SAFE_FOR_UNTRUSTED_CONTENT },
Index: toolkit/content/about.xhtml
===================================================================
--- toolkit/content/about.xhtml.orig
+++ toolkit/content/about.xhtml
@@ -55,27 +55,29 @@
 <head>
   <title>About:</title>
   <link rel="stylesheet" href="chrome://global/skin/about.css" type="text/css"/>
 </head>
 
 <body dir="&locale.dir;">
 <div id="aboutPageContainer">
   <div id="aboutLogoContainer">
-    <a href="http://www.mozilla.org/">
-      <img src="about:logo" alt="&brandShortName;"/>
-#expand <p id="version">&about.version; __MOZ_APP_VERSION__</p>
+    <a href="http://browser.garage.maemo.org/">
+      <img height="105" src="chrome://branding/content/about.png" alt="&brandShortName;"/>
     </a>
   </div>
+<p style="color:blue;font-size: 250%;" id="packname">&brandShortName;</p>
+#expand <p id="version">&about.version; __MOZ_APP_VERSION__</p>
 
   <ul id="aboutPageList">
     <li>&about.copy.beforeLink; <a href="about:credits">&about.copy.linkTitle;</a> &about.copy.afterLink;</li>
     <li>&about.license.beforeLink; <a href="about:license">&about.license.linkTitle;</a> &about.license.afterLink;</li>
     <li>&about.relnotes.beforeLink; <a id="releaseNotesURL" href="">&about.relnotes.linkTitle;</a> &about.relnotes.afterLink;</li>
     <li>&about.buildconfig.beforeLink; <a href="about:buildconfig">&about.buildconfig.linkTitle;</a> &about.buildconfig.afterLink;</li>
+    <li>&about.config.beforeLink; <a href="about:config">&about.config.linkTitle;</a>&about.config.afterLink;</li>
     <script type="application/javascript">
       // get release notes URL from prefs
       try {
         var formatter = Components.classes["@mozilla.org/toolkit/URLFormatterService;1"]
                                   .getService(Components.interfaces.nsIURLFormatter);
         var releaseNotesURL = formatter.formatURLPref("app.releaseNotesURL");
         var relnotes = document.getElementById("releaseNotesURL");
         relnotes.setAttribute("href", releaseNotesURL);
@@ -84,17 +86,17 @@
 
       var ua = navigator.userAgent;
       if (ua) {
         var list = document.getElementById("aboutPageList");
         var listItem = list.appendChild(document.createElement("li"));
         listItem.appendChild(document.createTextNode("&about.buildIdentifier;"));
         // replace the shortened 8-digit build id in the UA with the full 10-digit 
         // build id from navigator.buildID
-        var newUA = ua.replace(/\/\d{8}/, "/" + navigator.buildID);
-        listItem.appendChild(document.createTextNode(newUA));
+        //var newUA = ua.replace(/\/\d{8}/, "/" + navigator.buildID);
+        listItem.appendChild(document.createTextNode(ua));
       }
     </script>
   </ul>
 
 </div>
 </body>
 </html>
Index: toolkit/themes/winstripe/global/about.css
===================================================================
--- toolkit/themes/winstripe/global/about.css.orig
+++ toolkit/themes/winstripe/global/about.css
@@ -1,27 +1,26 @@
 html {
   background: -moz-Dialog;
 }
 
 body {
   margin: 0;
-  padding: 0 1em;
+  padding: 0 0em;
   color: -moz-FieldText;
   font: message-box;
 }
 
 #aboutPageContainer {
   position: relative;
   min-width: 330px;
   max-width: 50em;
   margin: 4em auto;
   border: 1px solid ThreeDShadow;
   -moz-border-radius: 10px;
-  padding: 3em;
   -moz-padding-start: 30px;
   background: -moz-Field;
 }
 
 #aboutLogoContainer {
   border: 1px solid ThreeDLightShadow;
   width: 300px;
   margin-bottom: 2em;
@@ -42,8 +41,12 @@
   -moz-margin-start: 1.5em;
   padding: 0;
   list-style: square;
 }
 
 ul > li {
   margin-top: .5em;
 }
+
+*:not(body):not(html) {
+  margin: 1% ! important;
+}
Index: toolkit/themes/winstripe/global/netError.css
===================================================================
--- toolkit/themes/winstripe/global/netError.css.orig
+++ toolkit/themes/winstripe/global/netError.css
@@ -8,17 +8,17 @@
  */
 
 html {
   background: -moz-Dialog;
 }
 
 body {
   margin: 0;
-  padding: 0 1em;
+  padding: 0 0em;
   color: -moz-FieldText;
   font: message-box;
 }
 
 h1 {
   margin: 0 0 .6em 0;
   border-bottom: 1px solid ThreeDLightShadow;
   font-size: 160%;
@@ -37,35 +37,35 @@
 ul {
   list-style: square;
 }
 
 #errorPageContainer {
   position: relative;
   min-width: 13em;
   max-width: 52em;
-  margin: 4em auto;
+  margin: 1em auto;
   border: 1px solid ThreeDShadow;
   -moz-border-radius: 10px;
-  padding: 3em;
-  -moz-padding-start: 30px;
+  padding: 1em;
+  -moz-padding-start: 5px;
   background: url("chrome://global/skin/icons/warning-large.png") left 0 no-repeat -moz-Field;
   -moz-background-origin: content;
 }
 
 body[dir="rtl"] #errorPageContainer {
   background-position: right 0;
 }
 
 #errorTitle {
-  -moz-margin-start: 80px;
+  -moz-margin-start: 40px;
 }
 
 #errorLongContent {
-  -moz-margin-start: 80px;
+  -moz-margin-start: 40px;
 }
 
 #errorShortDesc > p {
   overflow: auto;
   border-bottom: 1px solid ThreeDLightShadow;
   padding-bottom: 1em;
   font-size: 130%;
 }
Index: toolkit/themes/pinstripe/global/netError.css
===================================================================
--- toolkit/themes/pinstripe/global/netError.css.orig
+++ toolkit/themes/pinstripe/global/netError.css
@@ -8,17 +8,17 @@
  */
 
 html {
   background: -moz-Dialog;
 }
 
 body {
   margin: 0;
-  padding: 0 1em;
+  padding: 0 0em;
   color: -moz-FieldText;
   font: message-box;
 }
 
 h1 {
   margin: 0 0 .6em 0;
   border-bottom: 1px solid ThreeDLightShadow;
   font-size: 160%;
@@ -37,35 +37,35 @@
 ul {
   list-style: square;
 }
 
 #errorPageContainer {
   position: relative;
   min-width: 13em;
   max-width: 52em;
-  margin: 4em auto;
+  margin: 1em auto;
   border: 1px solid ThreeDShadow;
   -moz-border-radius: 10px;
-  padding: 3em;
-  -moz-padding-start: 30px;
+  padding: 1em;
+  -moz-padding-start: 5px;
   background: url("chrome://global/skin/icons/warning-large.png") left 0 no-repeat -moz-Field;
   -moz-background-origin: content;
 }
 
 body[dir="rtl"] #errorPageContainer {
   background-position: right 0;
 }
 
 #errorTitle {
-  -moz-margin-start: 80px;
+  -moz-margin-start: 40px;
 }
 
 #errorLongContent {
-  -moz-margin-start: 80px;
+  -moz-margin-start: 40px;
 }
 
 #errorShortDesc > p {
   overflow: auto;
   border-bottom: 1px solid ThreeDLightShadow;
   padding-bottom: 1em;
   font-size: 130%;
 }
Index: themes/classic/global/netError.css
===================================================================
--- themes/classic/global/netError.css.orig
+++ themes/classic/global/netError.css
@@ -8,17 +8,17 @@
  */
 
 html {
   background: -moz-Dialog;
 }
 
 body {
   margin: 0;
-  padding: 0 1em;
+  padding: 0 0em;
   color: -moz-FieldText;
   font: message-box;
 }
 
 h1 {
   margin: 0 0 .6em 0;
   border-bottom: 1px solid ThreeDLightShadow;
   font-size: 160%;
@@ -37,35 +37,35 @@
 ul {
   list-style: square;
 }
 
 #errorPageContainer {
   position: relative;
   min-width: 13em;
   max-width: 52em;
-  margin: 4em auto;
+  margin: 1em auto;
   border: 1px solid ThreeDShadow;
   -moz-border-radius: 10px;
-  padding: 3em;
-  -moz-padding-start: 30px;
-  background: url("chrome://global/skin/icons/alert-exclam.gif") left 0 no-repeat -moz-Field;
+  padding: 1em;
+  -moz-padding-start: 5px;
+  background: url("chrome://global/skin/icons/warning-large.png") left 0 no-repeat -moz-Field;
   -moz-background-origin: content;
 }
 
 body[dir="rtl"] #errorPageContainer {
   background-position: right 0;
 }
 
 #errorTitle {
-  -moz-margin-start: 64px;
+  -moz-margin-start: 40px;
 }
 
 #errorLongContent {
-  -moz-margin-start: 64px;
+  -moz-margin-start: 40px;
 }
 
 #errorShortDesc > p {
   overflow: auto;
   border-bottom: 1px solid ThreeDLightShadow;
   padding-bottom: 1em;
   font-size: 130%;
 }
@@ -75,17 +75,17 @@
   font-size: 110%;
 }
 
 #errorLongDesc > p {
 }
 
 #errorTryAgain {
   margin-top: 2em;
-  -moz-margin-start: 64px;
+  -moz-margin-start: 80px;
 }
 
 #brand {
   position: absolute;
   right: 0;
   bottom: -1.5em;
   -moz-margin-end: 10px;
   opacity: .4;
--- toolkit/locales/en-US/chrome/global/about.dtd.orig	2007-08-29 13:59:24.000000000 +0300
+++ toolkit/locales/en-US/chrome/global/about.dtd	2007-08-29 13:59:35.000000000 +0300
@@ -12,6 +12,10 @@
 <!ENTITY about.relnotes.linkTitle     "release notes">
 <!ENTITY about.relnotes.afterLink     "for this version.">
 
+<!ENTITY about.config.beforeLink "Change this application's">
+<!ENTITY about.config.linkTitle "internal settings">
+<!ENTITY about.config.afterLink " (unsupported).">
+
 <!ENTITY about.buildconfig.beforeLink "See the">
 <!ENTITY about.buildconfig.linkTitle  "build configuration">
 <!ENTITY about.buildconfig.afterLink  "used for this version.">
